CoEpi/extensions/Array+Convenience.swift

```diff
@@ -11,4 +11,11 @@ extension Array where Element: Equatable {
         }
         return uniqueValues
     }
+
+    func deleteFirst(element: Element) -> [Element] {
+        guard let index = firstIndex(of: element) else { return self }
+        var mutSelf = self
```

Comment:
@i-schuetz This reimplements copy-on-write logic that Swift arrays already have: https://developer.apple.com/documentation/swift/array
Seems unnecessary at first glance. Is it needed?

Reply:
You mean that it can be implemented like this:

```swift
mutating func deleteFirst(element: Element) -> [Element] {
    guard let index = firstIndex(of: element) else { return self }
    remove(at: index)
    return self
}
```

And called like this?

```
private func removeAlertLocally(alert: Alert) {
    let alertsState = alertsStateSubject.value
    switch alertsState {
    case .success(var alerts): // <-- var instead of let
        let newAlerts = alerts.deleteFirst(element: alert)
        alertsStateSubject.accept(.success(data: newAlerts))
    default: break
    }
}
```
This works, but I personally prefer an immutable api (not requiring to use `var`). But open to discuss.

Reply:
Not sure how exactly it relates to copy on write. We could implement a mutating version, like this:

```swift
mutating func deleteFirst(element: Element) {
    guard let index = firstIndex(of: element) else { return }
    remove(at: index)
}
```

And call it like e.g. this:

```swift
private func removeAlertLocally(alert: Alert) {
    let alertsState = alertsStateSubject.value
    switch alertsState {
    case .success(var alerts):
        alerts.deleteFirst(element: alert)
        alertsStateSubject.accept(.success(data: alerts))
    default: break
    }
}
```
But I personally prefer an immutable api. Open to discuss.

Reply:
> Not sure how exactly it relates to copy on write. We could implement a mutating version, like this:
> ...
> But I personally prefer an immutable api. Open to discuss.

Yes, mutating version would be my first pick, but consistency is also important, so lets go with immutable approach.